### PR TITLE
Manta v3.4.0 Release

### PR DESCRIPTION
**Pull request by Garandor**
- Bump crate versions
- Add changelog
- Rebenchmark weights
- Fixes for #724 
   - forgotten Cargo.lock
   - Fixed benchmarking logic
   - Hardcode LeaveDelay to 28 instead of calculating from fraction
   - Add explicit call index to each extrinsic
   - Whitelist ParachainStaking::Round storage item
-  `pallet-author-inherent`:  Fixed benchmarking logic

[Benchmark run with storage benchmark result](https://github.com/Manta-Network/Manta/actions/runs/3114478903)

Signed-off-by: Adam Reif <Garandor@manta.network>

relates to #793 

## TODO
- [x] Merge #724 
- [x] Build Changelog
- [x] Recalculate Weights -> https://github.com/Manta-Network/Manta/actions/runs/3109955065

---

Before we can approve this PR for merge, please make sure that **all** the following items have been checked off:
- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Added **one** label out of the `L-` group to this PR
- [x] Added **one or more** labels from the `A-` and `C-` groups to this PR
- [x] Explicitly labelled `A-calamari`, `A-dolphin` and/or `A-manta` if your changes are meant for/impact either of these (CI depends on it)
- [x] This PR is targeted against the *current*  Milestone ( otherwise discuss if it can be added in time)
- [x] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

**Comment by Garandor** [comment]
Don't merge yet, have to rebenchmark one more time because i whitelisted `ParachainStaking::Round`

**Comment by ghzlatarev** [comment]
> Don't merge yet, have to rebenchmark one more time because i whitelisted `ParachainStaking::Round`

Can you note the difference , i'm curious if the results change significantly.

**Comment by Garandor** [comment]
> > Don't merge yet, have to rebenchmark one more time because i whitelisted `ParachainStaking::Round`
> 
> Can you note the difference , i'm curious if the results change significantly.

The absolute numbers don't change, but extrinsics that use it have a read less in the calculation, see
https://github.com/Manta-Network/Manta/pull/797/commits/1399f63bd505545f8d1c3661d6125f80dd367ad2#diff-3862bebd2b4cf066443c88be50841b1d2d4b7e646cd6394985c94f8aafe31d62R215

File: runtime/calamari/src/weights/pallet_timestamp.rs [hunk 1]
```diff
@@ -54,24 +54,24 @@ pub struct SubstrateWeight<T>(PhantomData<T>);
 impl<T: frame_system::Config> pallet_timestamp::WeightInfo for SubstrateWeight<T> {
     // Storage: Timestamp Now (r:1 w:1)
     fn set() -> Weight {
-        (7_882_000 as Weight)
+        (18_208_000 as Weight)
```
Comment (Garandor) [inline runtime/calamari/src/weights/pallet_timestamp.rs:57]: 2.5x increase in pallet_timestamp

File: runtime/calamari/src/weights/pallet_treasury.rs [hunk 2]
```diff
@@ -57,44 +57,44 @@ pub trait WeightInfo {
...
             .saturating_add(T::DbWeight::get().writes(1 as Weight))
     }
     // Storage: Treasury Approvals (r:1 w:1)
     // Storage: Treasury Proposals (r:2 w:2)
     // Storage: System Account (r:4 w:4)
     fn on_initialize_proposals(p: u32, ) -> Weight {
-        (33_647_000 as Weight)
-            // Standard Error: 36_000
-            .saturating_add((32_831_000 as Weight).saturating_mul(p as Weight))
+        (23_710_000 as Weight)
```
Comment (Garandor) [inline runtime/calamari/src/weights/pallet_treasury.rs:95]: 1/3 cheaper

File: runtime/calamari/src/weights/pallet_parachain_staking.rs [hunk 3]
```diff
@@ -407,323 +348,296 @@ impl<T: frame_system::Config> pallet_parachain_staking::WeightInfo for Substrate
...
-			.saturating_add((52_172_000 as Weight).saturating_mul(x as Weight))
-			// Standard Error: 2_000
-			.saturating_add((156_000 as Weight).saturating_mul(y as Weight))
-			.saturating_add(T::DbWeight::get().reads(195 as Weight))
-			.saturating_add(T::DbWeight::get().reads((3 as Weight).saturating_mul(x as Weight)))
-			.saturating_add(T::DbWeight::get().writes(188 as Weight))
-			.saturating_add(T::DbWeight::get().writes((1 as Weight).saturating_mul(x as Weight)))
-	}
+    fn round_transition_on_initialize(x: u32, _y: u32, ) -> Weight {
+        (1_631_970_000 as Weight)
```
Comment (Garandor) [inline runtime/calamari/src/weights/pallet_parachain_staking.rs:355]: doublecheck

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,27 +1,46 @@
-# CHANGELOG 
+# CHANGELOG
+
+## v3.4.0
+### Added
+-[\#745](https://github.com/Manta-Network/Manta/pull/745) Workflow to check for labels
```
Comment (Dengjianping) [inline CHANGELOG.md]: Add one space between `-` and `[`, or the markdown won't be parsed correctly.

Rendered page: https://github.com/Manta-Network/Manta/blob/1399f63bd505545f8d1c3661d6125f80dd367ad2/CHANGELOG.md

File: runtime/dolphin/src/lib.rs [hunk 5]
```diff
@@ -117,7 +117,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
     spec_name: create_runtime_str!("dolphin"),
     impl_name: create_runtime_str!("dolphin"),
     authoring_version: 2,
-    spec_version: 3300,
+    spec_version: 3400,
```
Comment (Garandor) [inline runtime/dolphin/src/lib.rs]: change to 3402

File: runtime/calamari/src/lib.rs [hunk 6]
```diff
@@ -123,7 +122,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
     spec_name: create_runtime_str!("calamari"),
     impl_name: create_runtime_str!("calamari"),
     authoring_version: 2,
-    spec_version: 3300,
+    spec_version: 3400,
```
Comment (Garandor) [inline runtime/calamari/src/lib.rs]: 3402

File: runtime/calamari/src/lib.rs [hunk 7]
```diff
@@ -123,7 +122,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
...
     impl_version: 1,
     apis: RUNTIME_API_VERSIONS,
     transaction_version: 7,
```
Comment (Garandor) [inline runtime/calamari/src/lib.rs]: 9

File: pallets/parachain-staking/src/benchmarks.rs [hunk 8]
```diff
@@ -619,7 +619,7 @@ benchmarks! {
             Pallet::<T>::delegation_scheduled_requests(&collator),
             vec![ScheduledRequest {
                 delegator: caller,
-                when_executable: 3,
+                when_executable: 1 + <<T as Config>::RevokeDelegationDelay as Get<u32>>::get(),
```
Comment (Garandor) [inline pallets/parachain-staking/src/benchmarks.rs:622]: change mock.rs to use the correct delay and fix the tests accordingly

File: CHANGELOG.md (hunk 4 above)
Comment (Apokalip) [reply CHANGELOG.md]: https://github.com/Manta-Network/dev-tools/pull/12 fixed it in the generator itself

File: pallets/parachain-staking/src/benchmarks.rs (hunk 8 above)
Comment (Garandor) [reply pallets/parachain-staking/src/benchmarks.rs:622]: tracked in #805

File: runtime/calamari/src/weights/pallet_parachain_staking.rs (hunk 3 above)
Comment (Garandor) [reply runtime/calamari/src/weights/pallet_parachain_staking.rs:355]: This means a round change takes ~2ms, which when compared to the old pallet_session::rotate_session() mechanism, which used the full block is a good net gain.
Also, while the constant part is more expensive than moonbeam, the part that scales with `x` is much cheaper - this seems to be a benchmarking difference.

File: runtime/calamari/src/weights/pallet_treasury.rs (hunk 2 above)
Comment (Garandor) [reply runtime/calamari/src/weights/pallet_treasury.rs:95]: image
just a benchmarking outlier

File: runtime/calamari/src/weights/pallet_timestamp.rs (hunk 1 above)
Comment (Garandor) [reply runtime/calamari/src/weights/pallet_timestamp.rs:57]: image

benchmarking outlier. Only executed once per block. Ignoring

File: CHANGELOG.md (hunk 4 above)
Comment (Garandor) [reply CHANGELOG.md]: Pushed the new changelog. [It renders fine](https://github.com/Manta-Network/Manta/blob/release-v3.4.0/CHANGELOG.md). need to merge https://github.com/Manta-Network/dev-tools/pull/12 for verification to succeed

File: runtime/calamari/src/lib.rs (hunk 7 above)
Comment (Garandor) [reply runtime/calamari/src/lib.rs]: Change checked in, waiting for  https://substrate.stackexchange.com/questions/4972/are-there-negative-consequences-to-bumping-transaction-version-by-more-than-on

**Comment by Garandor** [comment]
Benchmark run with storage benchmark result https://github.com/Manta-Network/Manta/actions/runs/3114478903

File: runtime/calamari/src/lib.rs [hunk 9]
```diff
@@ -123,10 +122,10 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
     spec_name: create_runtime_str!("calamari"),
     impl_name: create_runtime_str!("calamari"),
     authoring_version: 2,
-    spec_version: 3300,
+    spec_version: 3402,
```
Comment (Dengjianping) [inline runtime/calamari/src/lib.rs:125]: Not sure why 3400 -> 3402 and 7 -> 9.

Comment (Garandor) [reply runtime/calamari/src/lib.rs:125]: `Calamari Parachain Staging` network was upgraded before to `3400`/8 and then downgraded back to a v3.3.0 state using version `3401`/9

To apply this runtime upgrade to both mainnet and staging, we need to have this weird version as discussed in the planning meeting yesterday

File: runtime/calamari/src/lib.rs (hunk 7 above)
Comment (Garandor) [reply runtime/calamari/src/lib.rs]: Response is in https://substrate.stackexchange.com/a/4988/504

**Review by Dengjianping — APPROVED** [review]

**Review by stechu — APPROVED** [review]
I didn't see any significant issue except the weight outlier.

**Pull request merged**